beps/bep_attrs.rst

```diff
@@ -0,0 +1,119 @@
+:BEP: ??
+:Title: Padding files and extended file attributes 
+:Version: $Revision$
+:Last-Modified: $Date$
+:Author:  The 8472 <the8472.bep@infinite-source.de>
+:Status:  Draft
+:Type:    Standards Track
+:Content-Type: text/x-rst
+:Created: 05-Aug-2016
+:Post-History: 
+
+
+Padding files and extended file attributes
+==========================================
+
+This BEP specifies some additional file properties beyond those described those in BEP 3 [#BEP-3]_.
+
+
+Multi-file format
+
+.. parsed-literal::
+
+    {
+      "info":
+      {
+        "files":
+        {[
+          {
+            "attr": "phxl",
+            "sha1": *<20 bytes>*,
+            "symlink path": ["dir1", "dir2", "target.ext"],
+            ...
+          },
+          {
+            ...
+          }
+        ]},
+        ...
+      },
+      ...
+    }
+    
+Single-file format
+
+.. parsed-literal::
+
+
+    {
+      "info":
+      {
+        "attr": "hx",
+        "sha1": *<20 bytes>*,
+        ...
+      },
+      ...
+    }
+
+
+
+
+``attr``
+  A variable-length string. When present the characters each represent a file attribute. l = symlink, x = executable, h = hidden, p = padding file. Characters appear in no particular order and unknown characters should be ignored.
+  
+``sha1``
+  20 bytes. The SHA1 digest calculated over the contents of the file itself, without any additional padding. Can be used to aid file deduplication [#BEP-38]_.
```

Comment:
in fact, I wouldn't mind dropping this from the specification (but obviously I'm willing to be convinced otherwise). newer versions of libtorrent no longer supports generating the file hashes anymore. I also don't know how it would help file deduplication in a safe way. Sure, it can act as a hint that's verified (so can the filename and file size though). And the verification is really the complicated part anyway, especially if it supports files that have different alignment to their pieces. If not, verification is so cheap that a hint may not be that valuable.